File: Go/leetcode/2255.go
package Leetcode

func countPrefixes(words []string, s string) int {
	result := 0

	// 遍历 words 数组
	for _, word := range words {
		//if len(word) > len(s) {  // 这个判断条件是多余的
		//	continue
		//}
		//if len(word) < len(s) {  // 可以将if else合并
		//	if s[:len(word)] == word {
		//		result++
		//	}
		//} else {
		//	if s == word {
		//		result++
		//	}
		//}
		if len(word) <= len(s) && s[:len(word)] == word {
			result++
		}
	}
	return result
}

File: Python/leetcode/2255.py
from typing import List


class Solution:
    def countPrefixes(self, words: List[str], s: str) -> int:
        result = 0
        for word in words:
            # startswith() 方法用于检查字符串是否是以指定子字符串开头，
            # 如果是则返回 True，否则返回 False。
            # 如果参数 beg 和 end 指定值，则在指定范围内检查。
            if s.startswith(word):
                result += 1
        return result